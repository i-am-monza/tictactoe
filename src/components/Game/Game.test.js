import React from './node_modules/react';
import ReactDOM from './node_modules/react-dom';
import test_renderer  from './node_modules/react-test-renderer';
import { Game } from './game.jsx';

describe('test cases for the Game component', () => {
  test('test if this component is rendered', () => {
    ReactDOM.render(<Game></Game>, document.createElement("div"));
  });

  test('test snapshot version', () => {
    expect(test_renderer.create(<Game></Game>)).toMatchSnapshot();
  });
});