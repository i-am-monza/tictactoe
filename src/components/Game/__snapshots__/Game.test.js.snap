// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for the Game component test snapshot version 1`] = `
<main
  className="app row"
  title="tic tac toe"
>
  <section
    className="header col-12"
  >
    <header
      title="tic tac toe header"
    >
      <h1
        className="lead text-center md-1"
      >
        Tic Tac Toe
      </h1>
      <span
        className="redos"
      >
        redo's:     
        <cite
          data-testid="x"
          id="x"
        >
          X
        </cite>
         | 
        <cite
          data-testid="o"
          id="o"
        >
          O
        </cite>
      </span>
    </header>
  </section>
  <section
    className="board col-8"
    title="playing board"
  >
    <table
      className="table table-borderless mt-3"
    >
      <tbody
        data-testid="3x3 grid"
      >
        <tr>
          <td>
            <button
              id="0"
              onClick={[Function]}
              title="Button 1"
              type="button"
            />
          </td>
          <td>
            <button
              id="1"
              onClick={[Function]}
              title="Button 2"
              type="button"
            />
          </td>
          <td>
            <button
              id="2"
              onClick={[Function]}
              title="Button 3"
              type="button"
            />
          </td>
        </tr>
        <tr>
          <td>
            <button
              id="3"
              onClick={[Function]}
              title="Button 4"
              type="button"
            />
          </td>
          <td>
            <button
              id="4"
              onClick={[Function]}
              title="Button 5"
              type="button"
            />
          </td>
          <td>
            <button
              id="5"
              onClick={[Function]}
              title="Button 6"
              type="button"
            />
          </td>
        </tr>
        <tr>
          <td>
            <button
              id="6"
              onClick={[Function]}
              title="Button 7"
              type="button"
            />
          </td>
          <td>
            <button
              id="7"
              onClick={[Function]}
              title="Button 8"
              type="button"
            />
          </td>
          <td>
            <button
              id="8"
              onClick={[Function]}
              title="Button 9"
              type="button"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </section>
  <section
    className="dashboard col"
    title="progress dashboard"
  >
    <form
      title="Progress dashboard"
    >
      <span
        className="sub-head"
      >
        <button
          onClick={[Function]}
          title="Rewind button"
        >
          <i
            className="fa fa-undo"
            id="rewind"
          />
        </button>
        <label
          id="player"
          title="Player O's turn"
        >
          O
        </label>
      </span>
      <ol
        data-testid="log"
        id="log"
      />
    </form>
  </section>
</main>
`;
